src/interface/blas1_interface.hpp

```diff
@@ -165,16 +151,21 @@ typename sb_handle_t::event_t _sdsdot(
     sb_handle_t &sb_handle, index_t _N, float sb, container_0_t _vx,
     increment_t _incx, container_1_t _vy, increment_t _incy, container_2_t _rs,
     const typename sb_handle_t::event_t &_dependencies) {
-  typename sb_handle_t::event_t dot_event{};
-  auto rs = make_vector_view(_rs, static_cast<increment_t>(1),
-                             static_cast<index_t>(1));
-
-  dot_event =
-      internal::_dot(sb_handle, _N, _vx, _incx, _vy, _incy, _rs, _dependencies);
-  auto addOp = make_op<ScalarOp, AddOperator>(sb, rs);
-  auto assignOp2 = make_op<Assign>(rs, addOp);
-  auto ret2 = sb_handle.execute(assignOp2, dot_event);
-  return blas::concatenate_vectors(dot_event, ret2);
+  if (!_N) {
+    sb_handle.wait(_dependencies);
+    auto ret = blas::helper::copy_to_device(sb_handle.get_queue(), &sb, _rs, 1);
```

Comment:
The copy_to_device function definition assumes that source and destination containers have the same data type, so this copy introduces an error when `_rs` is of type `double`.

Reply:
Thanks for the highlight @muhammad-tanvir-1211, I've fixed the issue with a `reinterpret_cast` on the `sb` reference to match the output (float or double), even-though the oneMKL spec only assumes float types for the `sdsdot` operator _(so we could have disabled the double support for it all-together as well)._